lint-tasks.yml

```diff
@@ -0,0 +1,217 @@
+      - "src/.clang-format"
+      - "{{.TASKFILE}}"
+      - ".clang-format"
+    deps: ["cpp-configs", "venv"]
+    cmds:
+      - task: ":utils:clang-format"
+        vars:
+          FLAGS: "--dry-run"
+          SRC_PATHS:
+            ref: ".CPP_LINT_DIRS"
+          VENV_DIR: "{{.LINT_VENV_DIR}}"
+
+  cpp-format-fix:
+    sources: *cpp_format_src_files
+    deps: ["cpp-configs", "venv"]
+    cmds:
+      - task: ":utils:clang-format"
+        vars:
+          FLAGS: "-i"
+          SRC_PATHS:
+            ref: ".CPP_LINT_DIRS"
+          VENV_DIR: "{{.LINT_VENV_DIR}}"
+
+  cpp-static-check:
+    # Alias task to `cpp-static-fix` since we don't currently support automatic fixes.
+    # NOTE: clang-tidy does have the ability to fix some errors, but the fixes can be inaccurate.
+    # When we eventually determine which errors can be safely fixed, we'll allow clang-tidy to
+    # fix them.
+    aliases: ["cpp-static-fix"]
+    sources:
+      - "{{.LINT_VENV_CHECKSUM_FILE}}"
+      - "{{.CLP_FFI_PY_SRC_DIR}}/**/*.cpp"
+      - "{{.CLP_FFI_PY_SRC_DIR}}/**/*.h"
+      - "{{.CLP_FFI_PY_SRC_DIR}}/**/*.hpp"
+      - "{{.CLP_FFI_PY_CMAKE_CACHE}}"
+      - "{{.CLP_FFI_PY_COMPILE_COMMANDS_DB}}"
+      - "{{.TASKFILE}}"
+      - "Taskfile.yml"
+      - ".clang-tidy"
+    deps: [":config-cmake-project", "cpp-configs", "venv"]
+    cmds:
+      - task: ":utils:clang-tidy"
+        vars:
+          FLAGS: "--config-file=.clang-tidy -p {{.CLP_FFI_PY_COMPILE_COMMANDS_DB}}"
+          SRC_PATHS:
+            ref: ".CPP_LINT_DIRS"
+          VENV_DIR: "{{.LINT_VENV_DIR}}"
+
+  py-check:
+    cmds:
+      - task: "mypy"
+        vars:
+          SRC_DIRS:
+            ref: ".PYTHON_LINT_DIRS"
+      - task: "py"
+        vars:
+          DOCFORMATTER_FLAGS: "--check --diff"
+          BLACK_FLAGS: "--check"
+          RUFF_FLAGS: ""
+          SRC_DIRS:
+            ref: ".PYTHON_LINT_DIRS"
+
+  py-fix:
+    cmds:
+      - task: "py"
+        vars:
+          DOCFORMATTER_FLAGS: "-i"
+          BLACK_FLAGS: ""
+          RUFF_FLAGS: "--fix"
+          SRC_DIRS:
+            ref: ".PYTHON_LINT_DIRS"
+
+  yml:
+    aliases:
+      - "yml-check"
+      - "yml-fix"
+    deps: ["venv"]
+    cmds:
+      - |-
+        . "{{.LINT_VENV_DIR}}/bin/activate"
+        yamllint \
+          --config-file "tools/yscope-dev-utils/lint-configs/.yamllint.yml" \
+          --strict \
+          .gersemirc \
+          .github \
+          docs \
+          lint-tasks.yml \
+          src/.clang-format \
+          Taskfile.yml
+
+  cmake:
+    internal: true
+    requires:
+      vars: ["FLAGS"]
+    cmd: |-
+      . "{{.LINT_VENV_DIR}}/bin/activate"
+      gersemi {{.FLAGS}} CMakeLists.txt {{.CLP_FFI_PY_SRC_DIR}}
+
+  mypy:
+    internal: true
+    requires:
+      vars: ["SRC_DIRS"]
+    deps: ["venv"]
+    cmds:
+      - for:
+          var: "SRC_DIRS"
+        cmd: |-
+          . "{{.LINT_VENV_DIR}}/bin/activate"
+          mypy {{.ITEM}}
+
+  py:
```

Comment:
Can we use `dir` to enforce the task to run under the root directory, so that the linter configures specified in `pyproject.toml` can be automatically applied? In that case we don't need to specify black's line length explicitly